core/coretypes/include/coretypes/ctutils.h

```diff
@@ -436,4 +437,10 @@ TPtr callNotNull(TPtr ptr, TFunc func)
     return std::invoke(func, std::forward<TPtr>(ptr));
 }
 
+inline bool validateTypeName(ConstCharPtr typeName)
+{
+    const std::regex validatorRegex("^[a-zA-Z_]+[a-zA-Z0-9_]*$");
```

Comment:
Should we also allow hyphens?

Reply:
Using hyphens in the attribute name is an incorrect from what I got from googling and testing myself